packages/types/src/type/Vote.ts

```diff
@@ -2,41 +2,79 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import { isBoolean } from '@polkadot/util';
+import { isBoolean, isNumber, isObject, u8aToBn, isU8a, bnToU8a } from '@polkadot/util';
 
-import I8 from '../primitive/I8';
+import Conviction from './Conviction';
+import U8a from '../codec/U8a';
+import Boolean from '../primitive/Bool';
 
 /**
  * @name Vote
  * @description
  * A number of lock periods, plus a vote, one way or the other.
  */
-export default class Vote extends I8 {
+export default class Vote extends U8a {
+  private _aye: Boolean;
+  private _conviction?: Conviction;
+
   constructor (value?: any) {
-    super(Vote.decodeVote(value));
+    // decoded is just 1 byte
+    // Aye: Most Significant Bit
+    // Conviction: 0000 - 0101
+    const decoded = Vote.decodeVote(value);
+
+    super(decoded);
+
+    const msb = decoded[0] << 7;
+    const conviction = decoded[0] & ((1 << 2) - 1);
```

Comment:
this might work, but I need too much time to understand it.

trick: you can use a bit mask: `const convition = decoded[0] & 0b01111111`